src/stash/src/postgres.rs

```diff
@@ -114,17 +154,63 @@ impl Postgres {
     /// ```
     async fn transact<F, T>(&mut self, f: F) -> Result<T, StashError>
     where
-        F: for<'a> FnOnce(&'a mut Transaction) -> BoxFuture<'a, Result<T, StashError>>,
+        F: for<'a> Fn(&'a mut Transaction) -> BoxFuture<'a, Result<T, StashError>>,
```

Comment:
I like the idea of using `Fn` here, it means we can call it multiple times (like `FnMut`, but those calls can't mutate (unless they do interior mutability)

I would probably add docs about how the only thing the closure/future should do is call functions on the `Transaction`